### PR TITLE
{chem}[GCCcore/8.3.0] xtb v6.2.3

### PR DESCRIPTION
**Pull request by dplacencia**
I'm working with XTB software support team to add support to XTB with GCC. A new release is coming...

### CODE REVIEW

File: easybuild/easyconfigs/x/xtb/xtb-6.3.0-GCCcore-8.3.0.eb [hunk 1]
```diff
@@ -0,0 +1,40 @@
+easyblock = 'MesonNinja'
+
+name = 'xtb'
```
Comment (boegel) [inline easybuild/easyconfigs/x/xtb/xtb-6.3.0-GCCcore-8.3.0.eb]: @dplacencia Looks like the official software name is `xTB`, so we should stick to that?

Also, do let us know if you need help with making the test suite pass.

**Comment by dplacencia** [comment]
Dear Kenneth, In the next few days I will upload the easyconfig fixed from xTB, I made this pull request by mistake so we can ignore it. I am working together with the developers of xTB to achieve a release that does not give problems with the foss-2019a toolchain, they are helping me a lot in this. I will soon fix this, thanks for your attention. Best regards.

----- Mensaje original -----
De: Kenneth Hoste <notifications@github.com>
Para: easybuilders/easybuild-easyconfigs <easybuild-easyconfigs@noreply.github.com>
CC: dplacencia <dplacencia@uclv.cu>, Mention <mention@noreply.github.com>
Enviado: Sun, 08 Mar 2020 09:12:33 -0400 (CDT)
Asunto: Re: [easybuilders/easybuild-easyconfigs] {chem}[GCCcore/8.3.0] xtb v6.3.0 (#9993)

boegel requested changes on this pull request.



> @@ -0,0 +1,40 @@
+easyblock = 'MesonNinja'
+
+name = 'xtb'

@dplacencia Looks like the official software name is `xTB`, so we should stick to that?

Also, do let us know if you need help with making the test suite pass.

-- 
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub:
https://github.com/easybuilders/easybuild-easyconfigs/pull/9993#pullrequestreview-370829080
-- 
************************************************
Ing. Daniel Placencia Alvarez
HPC Systems Administrator
Universidad Central “Marta Abreu” de Las Villas
dplacencia@uclv.cu  
www.uclv.hpc.cu
HPC UCLV Soporte: soporteuclv@listas.hpc.cu
************************************************

**Comment by awvwgk** [comment]
I'm the maintainer of `xtb` and finally found this place. First, thanks for your afford adding an easyconfig file for `xtb`. To clarify the software is indeed called `xtb`, but the main method it is implementing is named xTB, short for extended tight binding.

Also the next upcoming stable release will be 6.2.3 not 6.3.0, the latter is still in beta stage and not available as source.

File: easybuild/easyconfigs/x/xtb/xtb-6.3.0-foss-2019a-pre.eb [hunk 2]
```diff
@@ -0,0 +1,39 @@
+easyblock = 'MesonNinja'
+
+name = 'xtb'
+version = '6.3.0'
```
Comment (awvwgk) [inline easybuild/easyconfigs/x/xtb/xtb-6.3.0-foss-2019a-pre.eb]: The current version is 6.2.2, the next will be 6.2.3, either way this version is wrong.

File: easybuild/easyconfigs/x/xtb/xtb-6.3.0-foss-2019a-pre.eb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+homepage = 'https://xtb-docs.readthedocs.io'
+description = """Semiempirical Extended Tight-Binding Program Package"""
```
Comment (awvwgk) [inline easybuild/easyconfigs/x/xtb/xtb-6.3.0-foss-2019a-pre.eb]: This description is inconsistent to the Intel build
https://github.com/easybuilders/easybuild-easyconfigs/blob/c2cd3a9bcd774928a913cccf4c28241a7ec699a6/easybuild/easyconfigs/x/xtb/xtb-6.2.2-hotfix-intel-2017b-Python-3.6.3.eb#L8

File: easybuild/easyconfigs/x/xtb/xtb-6.3.0-foss-2019a-pre.eb [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+# Waiting for new release with GCC suport
+# sources = ['v%(version_major_minor)s-pre.1.tar.gz']
+sources = ['master.zip']
+github_account='grimme-lab'
+source_urls = [GITHUB_LOWER_SOURCE]
+
+builddependencies = [
+    ('Meson', '0.51.2', '-Python-3.7.4', ('GCCcore', '8.3.0')),
+    ('Ninja', '1.9.0', '', ('GCCcore', '8.3.0')),
```
Comment (awvwgk) [inline easybuild/easyconfigs/x/xtb/xtb-6.3.0-foss-2019a-pre.eb]: Why do you use different versions here compared to the Intel build?
https://github.com/easybuilders/easybuild-easyconfigs/blob/c2cd3a9bcd774928a913cccf4c28241a7ec699a6/easybuild/easyconfigs/x/xtb/xtb-6.2.2-hotfix-intel-2017b-Python-3.6.3.eb#L19-L20

**Comment by dplacencia** [comment]
Hi @boegel i'm not sure what is failing. Bot said "No conflicts detected", nevertheless it fail!

File: easybuild/easyconfigs/x/xtb/xtb-6.3.0-GCCcore-8.3.0.eb (hunk 1 above)
Comment (dplacencia) [reply easybuild/easyconfigs/x/xtb/xtb-6.3.0-GCCcore-8.3.0.eb]: I am having problems with the test suite, a problem in the dependency graph is jumping. However, at the end it says "No conflicts detected"

**Comment by boegel** [comment]
@dplacencia The bot is asserting that there are no conflicts, that's failing because you're introducing conflicts. `eb --check-conflicts` is your friend:

```
$ eb --from-pr 9993 --check-conflicts 
Conflict found for dependencies of xtb-6.2.3-foss-2019a: GCCcore-8.3.0 vs GCCcore-8.2.0 
	GCCcore-8.3.0 as dep of: Autoconf-2.69-GCCcore-8.3.0, Automake-1.16.1-GCCcore-8.3.0, Autotools-20180311-GCCcore-8.3.0, Bison-3.3.2-GCCcore-8.3.0, GMP-6.1.2-GCCcore-8.3.0, M4-1.4.18-GCCcore-8.3.0, Meson-0.51.2-GCCcore-8.3.0-Python-3.7.4, Ninja-1.9.0-GCCcore-8.3.0, Python-3.7.4-GCCcore-8.3.0, SQLite-3.29.0-GCCcore-8.3.0, Tcl-8.6.9-GCCcore-8.3.0, XZ-5.2.4-GCCcore-8.3.0, binutils-2.32-GCCcore-8.3.0, bzip2-1.0.8-GCCcore-8.3.0, flex-2.6.4-GCCcore-8.3.0, help2man-1.47.8-GCCcore-8.3.0, libffi-3.2.1-GCCcore-8.3.0, libreadline-8.0-GCCcore-8.3.0, libtool-2.4.6-GCCcore-8.3.0, ncurses-6.1-GCCcore-8.3.0, xtb-6.2.3-foss-2019a, zlib-1.2.11-GCCcore-8.3.0
	GCCcore-8.2.0 as dep of: Autoconf-2.69-GCCcore-8.2.0, Automake-1.16.1-GCCcore-8.2.0, Autotools-20180311-GCCcore-8.2.0, Bison-3.0.5-GCCcore-8.2.0, FFTW-3.3.8-gompi-2019a, GCC-8.2.0-2.31.1, M4-1.4.18-GCCcore-8.2.0, OpenBLAS-0.3.5-GCC-8.2.0-2.31.1, OpenMPI-3.1.3-GCC-8.2.0-2.31.1, ScaLAPACK-2.0.2-gompi-2019a-OpenBLAS-0.3.5, XZ-5.2.4-GCCcore-8.2.0, binutils-2.31.1-GCCcore-8.2.0, flex-2.6.4-GCCcore-8.2.0, foss-2019a, gompi-2019a, help2man-1.47.7-GCCcore-8.2.0, hwloc-1.11.11-GCCcore-8.2.0, libpciaccess-0.14-GCCcore-8.2.0, libtool-2.4.6-GCCcore-8.2.0, libxml2-2.9.8-GCCcore-8.2.0, numactl-2.0.12-GCCcore-8.2.0, xorg-macros-1.19.2-GCCcore-8.2.0, xtb-6.2.3-foss-2019a, zlib-1.2.11-GCCcore-8.2.0
Conflict found for dependencies of xtb-6.2.3-foss-2019a: XZ-5.2.4-GCCcore-8.3.0 vs XZ-5.2.4-GCCcore-8.2.0 
	XZ-5.2.4-GCCcore-8.3.0 as dep of: Meson-0.51.2-GCCcore-8.3.0-Python-3.7.4, Ninja-1.9.0-GCCcore-8.3.0, Python-3.7.4-GCCcore-8.3.0, xtb-6.2.3-foss-2019a
	XZ-5.2.4-GCCcore-8.2.0 as dep of: FFTW-3.3.8-gompi-2019a, OpenMPI-3.1.3-GCC-8.2.0-2.31.1, ScaLAPACK-2.0.2-gompi-2019a-OpenBLAS-0.3.5, foss-2019a, gompi-2019a, hwloc-1.11.11-GCCcore-8.2.0, libxml2-2.9.8-GCCcore-8.2.0, xtb-6.2.3-foss-2019a
Conflict found for dependencies of xtb-6.2.3-foss-2019a: binutils-2.32-GCCcore-8.3.0 vs binutils-2.31.1-GCCcore-8.2.0 
	binutils-2.32-GCCcore-8.3.0 as dep of: Autoconf-2.69-GCCcore-8.3.0, Automake-1.16.1-GCCcore-8.3.0, GMP-6.1.2-GCCcore-8.3.0, Meson-0.51.2-GCCcore-8.3.0-Python-3.7.4, Ninja-1.9.0-GCCcore-8.3.0, Python-3.7.4-GCCcore-8.3.0, SQLite-3.29.0-GCCcore-8.3.0, Tcl-8.6.9-GCCcore-8.3.0, XZ-5.2.4-GCCcore-8.3.0, bzip2-1.0.8-GCCcore-8.3.0, libffi-3.2.1-GCCcore-8.3.0, libreadline-8.0-GCCcore-8.3.0, libtool-2.4.6-GCCcore-8.3.0, ncurses-6.1-GCCcore-8.3.0, xtb-6.2.3-foss-2019a
	binutils-2.31.1-GCCcore-8.2.0 as dep of: Autoconf-2.69-GCCcore-8.2.0, Automake-1.16.1-GCCcore-8.2.0, FFTW-3.3.8-gompi-2019a, GCC-8.2.0-2.31.1, OpenBLAS-0.3.5-GCC-8.2.0-2.31.1, OpenMPI-3.1.3-GCC-8.2.0-2.31.1, ScaLAPACK-2.0.2-gompi-2019a-OpenBLAS-0.3.5, XZ-5.2.4-GCCcore-8.2.0, foss-2019a, gompi-2019a, hwloc-1.11.11-GCCcore-8.2.0, libpciaccess-0.14-GCCcore-8.2.0, libtool-2.4.6-GCCcore-8.2.0, libxml2-2.9.8-GCCcore-8.2.0, numactl-2.0.12-GCCcore-8.2.0, xorg-macros-1.19.2-GCCcore-8.2.0, xtb-6.2.3-foss-2019a
Conflict found for dependencies of xtb-6.2.3-foss-2019a: zlib-1.2.11-GCCcore-8.3.0 vs zlib-1.2.11-GCCcore-8.2.0 
	zlib-1.2.11-GCCcore-8.3.0 as dep of: Autoconf-2.69-GCCcore-8.3.0, Automake-1.16.1-GCCcore-8.3.0, GMP-6.1.2-GCCcore-8.3.0, Meson-0.51.2-GCCcore-8.3.0-Python-3.7.4, Ninja-1.9.0-GCCcore-8.3.0, Python-3.7.4-GCCcore-8.3.0, SQLite-3.29.0-GCCcore-8.3.0, Tcl-8.6.9-GCCcore-8.3.0, XZ-5.2.4-GCCcore-8.3.0, binutils-2.32-GCCcore-8.3.0, bzip2-1.0.8-GCCcore-8.3.0, libffi-3.2.1-GCCcore-8.3.0, libreadline-8.0-GCCcore-8.3.0, libtool-2.4.6-GCCcore-8.3.0, ncurses-6.1-GCCcore-8.3.0, xtb-6.2.3-foss-2019a
	zlib-1.2.11-GCCcore-8.2.0 as dep of: Autoconf-2.69-GCCcore-8.2.0, Automake-1.16.1-GCCcore-8.2.0, FFTW-3.3.8-gompi-2019a, GCC-8.2.0-2.31.1, OpenBLAS-0.3.5-GCC-8.2.0-2.31.1, OpenMPI-3.1.3-GCC-8.2.0-2.31.1, ScaLAPACK-2.0.2-gompi-2019a-OpenBLAS-0.3.5, XZ-5.2.4-GCCcore-8.2.0, binutils-2.31.1-GCCcore-8.2.0, foss-2019a, gompi-2019a, hwloc-1.11.11-GCCcore-8.2.0, libpciaccess-0.14-GCCcore-8.2.0, libtool-2.4.6-GCCcore-8.2.0, libxml2-2.9.8-GCCcore-8.2.0, numactl-2.0.12-GCCcore-8.2.0, xorg-macros-1.19.2-GCCcore-8.2.0, xtb-6.2.3-foss-2019a
ERROR: One or more conflicts detected!
```

File: easybuild/easyconfigs/x/xtb/xtb-6.2.3-foss-2019a.eb [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+toolchain = {'name': 'foss', 'version': '2019a'}
+
+source_urls = [GITHUB_LOWER_SOURCE]
+sources = [{'download_filename': 'v%(version)s.tar.gz', 'filename': SOURCE_TAR_GZ}]
+checksums = ['620dfb93aa07fd5144d25134613a5d7f4c1fa4fb5070c30768a6335c8c0b4187']
+
+github_account = 'grimme-lab'
+builddependencies = [
+    ('Meson', '0.51.2', '-Python-3.7.4', ('GCCcore', '8.3.0')),
+    ('Ninja', '1.9.0', '', ('GCCcore', '8.3.0')),
```
Comment (boegel) [inline easybuild/easyconfigs/x/xtb/xtb-6.2.3-foss-2019a.eb]: @dplacencia You should not hardcode the toolchain here for these dependencies. This is also why the tests are failing: `foss/2019b` is based on `GCCcore/8.2.0`, while you're forcing the use of `GCCcore/8.3.0` here.

If you want to use Python 3.7.4, you should update to the toolchain `foss/2019b`:

```python
toolchain = {'name': 'foss', 'version': '2019b'}
...
dependencies = [
    ('Meson', '0.51.2', '-Python-3.7.4'),
    ('Ninja', '1.9.0'),
]
```

If you prefer sticking to `foss/2019a`, you'll need to adjust the line for `Meson`:

```python
('Meson', '0.50.0', '-Python-3.7.2'),

**Comment by lexming** [comment]
Test report by @lexming
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
login2.cerberus.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) Gold 6126 CPU @ 2.60GHz, Python 2.7.5
See https://gist.github.com/2a4e9040e60250707de3faf2d27e3df0 for a full test report.

File: easybuild/easyconfigs/x/xtb/xtb-6.2.3-foss-2019b.eb [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+configopts = "-Dla_backend='custom' "
+configopts += "-Dcustom_libraries='openblas' "
+configopts += "--optimization=2 "
+configopts += "--buildtype release "
+configopts += "--warnlevel 0 "
+configopts += "-Dpython=false"
+
+runtest = 'meson'
+testopts = 'test -C %(builddir)s/easybuild_obj -t 10'  # Ensure test don't timeout
```
Comment (lexming) [inline easybuild/easyconfigs/x/xtb/xtb-6.2.3-foss-2019b.eb]: My previous test failed due to 6 tests timing out. I had to increase the time limit to 60 for it to work.
```suggestion
testopts = 'test -C %(builddir)s/easybuild_obj -t 60'  # Ensure test don't timeout
```

File: easybuild/easyconfigs/x/xtb/xtb-6.2.3-foss-2019b.eb [hunk 7]
```diff
@@ -0,0 +1,41 @@
+easyblock = 'MesonNinja'
+
+name = 'xtb'
```
Comment (lexming) [inline easybuild/easyconfigs/x/xtb/xtb-6.2.3-foss-2019b.eb]: Please, change name capitalization here and in the file name to match official name. 
```suggestion
name = 'xTB'
```

Comment (awvwgk) [reply easybuild/easyconfigs/x/xtb/xtb-6.2.3-foss-2019b.eb]: Please, the name of the program is still `xtb`.

Comment (lexming) [reply easybuild/easyconfigs/x/xtb/xtb-6.2.3-foss-2019b.eb]: No problem, the name can stay as `xtb`. We already have older easyconfigs for it with that name as well.

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
login2.cerberus.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) Gold 6126 CPU @ 2.60GHz, Python 2.7.5
See https://gist.github.com/154261c3ed66a954c0df56660930bec8 for a full test report.

File: easybuild/easyconfigs/x/xtb/xtb-6.2.3-foss-2019b.eb [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+configopts += "--buildtype release "
+configopts += "--warnlevel 0 "
+configopts += "-Dpython=false"
+
+runtest = 'meson'
+testopts = 'test -C %(builddir)s/easybuild_obj -t 60'  # Ensure test don't timeout
+
+sanity_check_paths = {
+    'files': ['bin/xtb'],
+    'dirs': ['bin', 'include', 'lib', 'share'],
```
Comment (lexming) [inline easybuild/easyconfigs/x/xtb/xtb-6.2.3-foss-2019b.eb]: Now that the test passed, I have one final request to ensure that sanity checks does not miss any failed builds. It is preferable to check for files of headers and libraries as well. See below
```suggestion
    'files': ['bin/xtb', 'include/xtb/xtb.h'] + ['lib/libxtb.%s' % e for e in ('a', SHLIB_EXT)],
    'dirs': ['share'],
```

Comment (dplacencia) [reply easybuild/easyconfigs/x/xtb/xtb-6.2.3-foss-2019b.eb]: I agree

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
login2.cerberus.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) Gold 6126 CPU @ 2.60GHz, Python 2.7.5
See https://gist.github.com/a6b29c111cc440183a50a5922c477824 for a full test report.

**Comment by lexming** [comment]
Going in, thanks @dplacencia !

**Pull request merged**